doc/source/development/rfc/rfc103_schema_open_option.rst

```diff
@@ -0,0 +1,113 @@
+.. _rfc-103:
+
+===================================================================
+RFC 103 add a SCHEMA open option to selected OGR drivers
+===================================================================
+
+=============== =============================================
+Author:         Alessandro Pasotti
+Contact:        elpaso at itopen.it
+Status:         Draft
+Created:        2024-10-22
+=============== =============================================
+
+Summary
+-------
+
+This RFC enables users to specify a SCHEMA open option in the OGR
+drivers that support it.
+
+The new option will be used to override the auto-detected fields types.
+
+Motivation
+----------
+
+Several OGR drivers must guess the attribute data type: CSV, GeoJSON, SQLite,
+the auto-detected types are not always correct and the user has no way to
+override them at opening tim.
+
+For the details please see the discussion attached to the issue: https://github.com/OSGeo/gdal/issues/10943
+
+Implementation
+--------------
+
+A new open option named SCHEMA will be added to the following drivers:
```

Comment:
We should mention this open option will be a reserved one. If a driver uses it, it must be for that purpose

And I believe we should modify ogr2ogr so that -mapFieldType uses the capability offered by the SCHEMA open option when available